src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -120,12 +147,64 @@ def _detect_performance_issue(data: Event, sdk_span: Any):
         DetectorType.N_PLUS_ONE_SPANS: NPlusOneSpanDetector(detection_settings, data),
     }
 
+    # Create performance issues for duplicate spans first
+    used_perf_issue_detectors = {
+        DetectorType.DUPLICATE_SPANS_HASH: DuplicateSpanHashDetector(detection_settings, data),
+        DetectorType.DUPLICATE_SPANS: DuplicateSpanDetector(detection_settings, data),
+    }
+
     for span in spans:
         for _, detector in detectors.items():
             detector.visit_span(span)
 
     report_metrics_for_detectors(event_id, detectors, sdk_span)
 
+    detected_problems = [
+        (i, detector_type)
+        for detector_type, d in used_perf_issue_detectors.items()
+        for _, i in d.stored_issues
+    ]
+
+    truncated_problems = detected_problems[:PERFORMANCE_GROUP_COUNT_LIMIT]
+
+    formatted_perf_problems = [
+        prepare_problem_for_grouping(problem, data) for problem, _ in truncated_problems
+    ]
+    return formatted_perf_problems
+
+
+def prepare_problem_for_grouping(problem: PerformanceSpanProblem, data: Event):
+    transaction_name = data.get("transaction")
+    first_span_id = problem[2][0]
+    spans = data.get("spans", [])
+    first_span = next((span for span in spans if span["span_id"] == first_span_id), None)
+    op = first_span["op"]
+    hash = first_span["hash"]
+    desc = first_span["desc"]
+
+    # TODO map detectors to the group type enum
+    group_fingerprint = fingerprint_group(
+        transaction_name, op, hash, GroupType.PERFORMANCE_DUPLICATE_SPANS
+    )
+
+    prepared_problem = PerformanceProblem(
+        fingerprint=group_fingerprint,
+        op=op,
+        desc=desc,
+        type=GroupType.PERFORMANCE_DUPLICATE_SPANS,
+        spans_involved=problem[2],
+    )
+
+    return prepared_problem
+
+
+def fingerprint_group(transaction_name, span_op, hash, problem_class):
+    signature = (str(transaction_name) + str(span_op) + str(hash) + str(problem_class)).encode(
+        "utf-8"
+    )
+    full_fingerprint = hashlib.sha1(signature).hexdigest()
+    return full_fingerprint
```

Comment:
So if we have this fingerprint stored in an indexed way on a transaction row in clickhouse... we can test out the hit rate of `transaction_name` vs no-`transaction_name` very easily.

One other comment, I'd consider maybe taking the known cardinality strings and leave them as plain text, it will make looking at these values easier to interpret and be more human readable while we work on this (and queryable with string search if we want to do a bit of analysis). You can confirm the storage aspect w/ SnS, but it will remain a bounded-size string. 

eg. for `problem_class` since there are a known amount, we don't really need to collapse it into the the fingerprint.

```py
    signature = (str(transaction_name) + str(span_op) + str(hash)).encode(
        "utf-8"
    )
    full_fingerprint = hashlib.sha1(signature).hexdigest()
    return f"{problem_class}-{full_fingerprint}"
```

Instead of standing up a new column in theory you could prefix it with a version of our fingerprinting eg. 

```
f"1-{problem_class}-{full_fingerprint}"
```

which _does_ require a wildcard search on the groups, but for developing this feature (up until we potentially wipe that column before going live) it would let us test out different group fingerprint strategies pretty quickly by just incrementing a version to discern between changes (and using discover to check, say, transaction_name vs. no transaction_name)

Reply:
got it. opted in to using `f"1-{problem_class}-{full_fingerprint}"` 👍

Reply:
Just checked, forgot we were doing `group_ids` on transaction now, not fingerprints, so this isn't searchable regardless, let's just leave it and figure out if we need to change it later, good enough for now.